SDL_ttf.c

```diff
@@ -1366,28 +1366,81 @@ static SDL_INLINE int Render_Line(const render_mode_t render_mode, int subpixel,
 #endif
 }
 
-static SDL_Surface* Create_Surface_Solid(int width, int height, SDL_Color fg, Uint32 *color)
+#ifndef SIZE_MAX
+# define SIZE_MAX ((size_t) -1)
+#endif
+
+/* TODO: Use SDL_size_mul_overflow after it's in a release */
+SDL_FORCE_INLINE int size_mul_overflow (size_t a,
+                                        size_t b,
+                                        size_t *ret)
 {
-    const int alignment = Get_Alignement() - 1;
-    SDL_Surface *textbuf;
-    Sint64 size;
+#if defined(_SDL_HAS_BUILTIN) && _SDL_HAS_BUILTIN(__builtin_mul_overflow)
```

Comment:
Looks like this doesn't do what I wanted it to if `_SDL_HAS_BUILTIN` isn't defined, so it'll need more backwards-compat goo.

Reply:
How about just do the C versions of those functions and don't worry about the builtins, and use the SDL versions if built with a newer SDL?